CloudNotes/CloudNotes/DetailViewController.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  DetailViewController.swift
+//  CloudNotes
+//
+//  Created by Yeon on 2021/02/16.
+//
+
+import UIKit
+
+final class DetailViewController: UIViewController {
+    var memo: Memo? {
+        didSet {
+            refreshUI()
+        }
+    }
+    
+    private var memoBodyTextView: UITextView = {
+        let textView = UITextView()
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        textView.dataDetectorTypes = [.link, .phoneNumber, .calendarEvent]
+        textView.isSelectable = true
+        textView.isEditable = false
+        textView.isUserInteractionEnabled = true
+        textView.font = .preferredFont(forTextStyle: .body)
+        textView.textColor = .black
+        return textView
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTextView()
+        setupNavigationBar()
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        setupNavigationBar()
+    }
+    
+    private func setupNavigationBar() {
+        if traitCollection.userInterfaceIdiom == .pad &&
+            UIDevice.current.orientation.isLandscape {
+            navigationController?.navigationBar.isHidden = true
+        }
+        else {
+            navigationController?.navigationBar.isHidden = false
+        }
+    }
+    
+    private func setupTextView() {
+        setTapGesture()
+        view.addSubview(memoBodyTextView)
+        NSLayoutConstraint.activate([
+            memoBodyTextView.topAnchor.constraint(equalTo: view.topAnchor),
+            memoBodyTextView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            memoBodyTextView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            memoBodyTextView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+    
+    private func setTapGesture() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(tapTextView(_:)))
+        memoBodyTextView.addGestureRecognizer(tapGesture)
+    }
+    
+    private func refreshUI() {
+        loadViewIfNeeded()
+        guard let memo = memo else {
+            return
+        }
+        let title = memo.title
+        let fontAttributeKey = NSAttributedString.Key(rawValue: kCTFontAttributeName as String)
+        let fontSize = UIFont.preferredFont(forTextStyle: .title1)
+        let content = NSMutableAttributedString(string: "\(memo.title)\n\n\(memo.body)")
+        content.addAttribute(fontAttributeKey, value: fontSize, range: NSMakeRange(0, title.count))
```

Comment:
title 부분이 잘리고 겹치는 문제가 발생하는 건 NSAttributedString에 속성값이 잘못 설정되어서 그런 것 같아요
아래와 같이 설정해주는 건 어떨까요?

```suggestion
        let content = NSMutableAttributedString(string: "\(memo.title)\n\n\(memo.body)")
        content.addAttribute(NSAttributedString.Key.font, value: UIFont.systemFont(ofSize: 18.0), range: NSMakeRange(0, title.count))
```

Reply:
자세한 설명 감사합니다 delma!! 
그러나 저희가 알려주신 코드로 실행해보았는데, `UIFont.systemFont(ofSize: 18.0)`을 
`UIFont.preferredFont(forTextStyle: .title1)`로 변경해서 실행해 본 결과 처음과 같이 글자가 깨졌었습니다.

그래서 이유를 찾아낼 결과 content는 `memo.title`, `\n\n` ,`memo.body`로 구성되어있는데 
이때 memo.title의 font만 지정이 되고 나머지 줄바꿈과 memo.body의 font가 지정되지 않아서 오류가 발생했던 것 같습니다.

따라서 해당 부분의 코드에서 다음과 같이 뒷부분의 글자에도 font를 지정해주면 문제없이 해결되는 것을 확인하였습니다!!
```swift
let content = NSMutableAttributedString(string: "\(memo.title)\n\n\(memo.body)")
content.addAttribute(NSAttributedString.Key.font, value: UIFont.preferredFont(forTextStyle: .title1), range: NSMakeRange(0, title.count))
content.addAttribute(NSAttributedString.Key.font, value: UIFont.preferredFont(forTextStyle: .title1), range: NSMakeRange(title, memo.body.count))

memoBodyTextView.attributedText = content
```

그리고 또 다른 구현방식도 사용해보았습니다.
```swift
let content = NSMutableAttributedString(string: memo.title, attributes: [NSAttributedString.Key.font: UIFont.preferredFont(forTextStyle: .title1)])
content.append(NSAttributedString(string: "\n\n" + memo.body, attributes: [NSAttributedString.Key.font: UIFont.preferredFont(forTextStyle: .body)]))

memoBodyTextView.attributedText = content
```